lib/engine/game/g_1822_africa/game.rb

```diff
@@ -292,57 +387,128 @@ def timeline
           timeline = []
 
           companies = bank_companies.map do |company|
-            "#{self.class::COMPANY_SHORT_NAMES[company.id]}#{'*' if bidbox_minors.any? { |c| c == company }}"
+            "#{self.class::COMPANY_SHORT_NAMES[company.id]}#{'*' if bidbox.any? { |c| c == company }}"
           end
 
           timeline << companies.join(', ') unless companies.empty?
 
           timeline
         end
 
-        def bidbox_minors_refill!
-          @bidbox_minors_cache = bank_companies
-                                   .first(self.class::BIDDING_BOX_MINOR_COUNT)
-                                   .select { |c| c.id[0] == self.class::COMPANY_MINOR_PREFIX }
-                                   .map(&:id)
+        def unowned_purchasable_companies(_entity)
+          bank_companies
+        end
+
+        def bidbox_refill!
+          @bidbox_cache = bank_companies
+                                        .first(self.class::BIDDING_BOX_MINOR_COUNT)
+                                        .select { |c| minor?(c) }
+                                        .map(&:id)
 
           # Set the reservation color of all the minors in the bid boxes
-          @bidbox_minors_cache.each do |company_id|
+          @bidbox_cache.each do |company_id|
             corporation_by_id(company_id[1..-1]).reservation_color = self.class::BIDDING_BOX_MINOR_COLOR
           end
+
+          @bidbox_companies_size = bidbox.length
+        end
+
+        def init_stock_market
+          G1822Africa::StockMarket.new(game_market, [])
+        end
+
+        def operating_round(round_num)
+          Engine::Round::Operating.new(self, [
+            G1822::Step::PendingToken,
+            G1822::Step::FirstTurnHousekeeping,
+            Engine::Step::AcquireCompany,
+            G1822::Step::DiscardTrain,
+            G1822::Step::SpecialChoose,
+            G1822::Step::SpecialTrack,
+            G1822::Step::SpecialToken,
+            G1822::Step::Track,
+            G1822::Step::DestinationToken,
+            G1822::Step::Token,
+            G1822::Step::Route,
+            G1822Africa::Step::Dividend,
+            G1822::Step::BuyTrain,
+            G1822Africa::Step::MinorAcquisition,
+            G1822::Step::PendingToken,
+            G1822::Step::DiscardTrain,
+            G1822::Step::IssueShares,
+          ], round_num: round_num)
+        end
+
+        def stock_round(round_num = 1)
+          G1822Africa::Round::Stock.new(self, [
+            Engine::Step::DiscardTrain,
+            G1822::Step::BuySellParShares,
+          ], round_num: round_num)
+        end
+
+        def total_rounds(name)
+          @operating_rounds if name == self.class::OPERATING_ROUND_NAME
+          2 if name == self.class::STOCK_ROUND_NAME
```

Comment:
In ORs this method will return `nil`.

This should use a `case name` block. https://rubystyle.guide/#case-vs-if-else

Reply:
Sorry for this silly mistake! I learned some  Ruby specifically to contribute here so I may miss some conventions or make mistakes :)

Reply:
If you do ever have a function similar to this but the `if`s are checking different conditions, what you have would be fine if the first line just started with a `return`. You could make both/all lines start with a `return` and let rubocop clean up the last `return`.